### PR TITLE
Add video upload support

### PR DESCRIPTION
**Pull request by fx**
Adds media/upload support for chunked video uploads.

See: 
https://dev.twitter.com/rest/public/uploading-media
https://blog.twitter.com/2015/rest-api-now-supports-native-video-upload

### CODE REVIEW

File: lib/twitter/rest/tweets.rb [hunk 1]
```diff
@@ -300,6 +296,45 @@ def retweeters_ids(*args)
 
     private
 
+      # Uploads images and videos. Videos require multiple requests and uploads in chunks of 5 Megabytes.
+      # The only supported video format is mp4.
+      #
+      # @see https://dev.twitter.com/rest/public/uploading-media
+      def upload(media) # rubocop:disable MethodLength, AbcSize
+        if !(File.basename(media) =~ /\.mp4$/)
```
Comment (filipegiusti) [inline lib/twitter/rest/tweets.rb]: I store my videos on S3 using refile, thus they don't have a `.mp4` basename, I guess checking for the basename to determine if something is a video isn't the best way to do it.

**Comment by stoneG** [comment]
Is there an ETA on when this will be merged?

**Comment by pietrop** [comment]
While I wait for this to be merged with the main repo, and some documentation being written about it, I was wondering if working with the `video_support` branch of this fork, is the following code how you would upload a video with a tweet? 

```
require "twitter"
require "twitter_oauth"

client = Twitter::REST::Client.new do |config|
  config.consumer_key        = YOUR_CONSUMER_KEY
  config.consumer_secret     = YOUR_CONSUMER_SECRET
  config.access_token        = YOUR_ACCESS_TOKEN
  config.access_token_secret = YOUR_ACCESS_SECRET
end

## Not working
client.update_with_media("testing twitter api with video upload", File.new("./test.mp4"))
```

**Comment by pietrop** [comment]
is the following code how you would upload a video using the `video_support` branch of this fork?
```ruby
require "twitter"
require "open-uri"
require "twitter_oauth"

client = Twitter::REST::Client.new do |config|
  config.consumer_key        = YOUR_CONSUMER_KEY
  config.consumer_secret     = YOUR_CONSUMER_SECRET
  config.access_token        = YOUR_ACCESS_TOKEN
  config.access_token_secret = YOUR_ACCESS_SECRET
end

## Not working
client.update_with_media("testing twitter api with video upload", File.new("./test.mp4"))
```

File: lib/twitter/rest/tweets.rb (hunk 1 above)
Comment (fx) [inline lib/twitter/rest/tweets.rb]: True, didn't want to add any dependencies or complexity for this here though, iirc.

**Comment by fx** [comment]
@pietrop yes

**Comment by PuKoren** [comment]
+1

**Comment by GaelleG** [comment]
+1

**Comment by sferik** [comment]
This branch needs to be rebased before it can be merged.

**Comment by pietrop** [comment]
Sorry, what do you mean by rebase?

**Comment by sferik** [comment]
@pietrop This patch can no longer be applied to `master`, since `master` has changed. There’s lots of documentation on how to rebase in Git but the process is essentially:

1. creating a new branch from the current `master`,
2. applying this patch to that branch, and
3. resolving any conflicts that may have occurred along the way.

**Comment by pietrop** [comment]
ahh, ok I see, thanks for clarifying @sferik

**Comment by fx** [comment]
Opening a new pull request.

**Comment by narlei** [comment]
Why only supports MP4 video file? There any way to send other formats?

**Pull request closed without merging**